### PR TITLE
Improve Criminal Records Computer usability 

### LINKED ISSUE #30087: Clearing multiple people of wanted status is awful.
## Description

Ninja hacks Criminal records computer and sets everyone to wanted. Begin to go through to try and reset everyone. Scream at it being a horrible mess.

**Reproduction**

Scroll through computer and set someone to no longer wanted
scroll down to select next person. It jumps around and resets again.

**Screenshots**

https://github.com/user-attachments/assets/542cef30-7a70-4305-906a-949e3170abfa

**Additional context**

### PR DESCRIPTION
**Pull request by eoineoineoin**
## About the PR

Fixes #30087 - UI jumped around every time it was updated, including whenever a record was changed or selected.

## Why / Balance

Bad UIs are bad for fun.

## Technical details

Classic case of "deleting and recreating all the elements in the UI."

The CriminalRecordsConsoleWindow is a large copy-paste of the GeneralStationRecordConsoleWindow, which also has the same problem, but I didn't touch that, as that UI doesn't actually do anything, so I don't think it's used?

If desired, I can look into refactoring both to share code, but won't have any free time to do this in the next two weeks.

## Media

Before (jump city):

https://github.com/user-attachments/assets/fc9d1d91-cd60-4421-b733-63e1c6406d94

After (smooth sailing):

https://github.com/user-attachments/assets/cbc00a60-ea63-4685-aac3-905bc4bdc1b8

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by Emisse** [comment]
linq scary needs linqer to look at it

**Comment by mach-mikulas** [comment]
filtering crewmembers by criminal status would be good addition to criminal records computer

**Comment by JIPDawg** [comment]
> filtering crewmembers by criminal status would be good addition to criminal records computer

I agree, but maybe in a future PR. I just want this one merged asap.

File: Content.Client/CriminalRecords/CriminalRecordsConsoleWindow.xaml.cs [hunk 1]
```diff
@@ -167,19 +162,58 @@ public void UpdateState(CriminalRecordsConsoleState state)
...
+    private void PopulateRecordListing(Dictionary<uint, string>? listing)
     {
-        _isPopulating = true;
+        if (listing == null)
+        {
+            RecordListing.Clear();
+        }
 
-        foreach (var (key, name) in listing)
+        var entries = listing!.ToList();
```
Comment (metalgearsloth) [inline Content.Client/CriminalRecords/CriminalRecordsConsoleWindow.xaml.cs:172]: Allowing null inputs then null suppressing is... interesting.

Going to checkout this pr but personally I don't like taking null input adding surprise changes to the method.

**Comment by metalgearsloth** [comment]
scary pr

**Pull request merged**